static/js/components/dashboard/courses/util.js

```diff
@@ -0,0 +1,80 @@
+// @flow
+import moment from 'moment';
+import R from 'ramda';
+
+import type { CourseRun } from '../../../flow/programTypes';
+import {
+  STATUS_OFFERED,
+  STATUS_PASSED,
+  STATUS_MISSED_DEADLINE,
+  STATUS_NOT_PASSED,
+  STATUS_PENDING_ENROLLMENT,
+} from '../../../constants';
+import { S } from '../../../lib/sanctuary';
+
+export const courseStartDateMessage = (courseRun: CourseRun) => {
+  if (R.isNil(courseRun.course_start_date)) {
+    return "";
+  }
+
+  let startDate = moment(courseRun.course_start_date);
+  let daysUntilStart = startDate.diff(moment(), 'days');
+  if (daysUntilStart === 0) {
+    return "Course starts today";
+  } else if (daysUntilStart < 0) {
+    return `Course started ${startDate.format('M/D/YYYY')}`;
+  } else if (daysUntilStart < 10) {
+    return `Course starts in ${daysUntilStart} days`;
+  } else {
+    return `Course starts ${startDate.format('M/D/YYYY')}`;
+  }
+};
+
+export const hasPearsonExam = R.propEq('has_exam', true);
+
+export const userIsEnrolled = R.compose(
+  R.not,
+  R.contains(R.__, [STATUS_OFFERED, STATUS_PENDING_ENROLLMENT]),
```

Comment:
I feel like this would be better represented as a whitelist instead of a blacklist. Also, what about `STATUS_PAID_BUT_NOT_ENROLLED`?